packages/paste-website/plugins/remark-headings-plugin.mjs

```diff
@@ -0,0 +1,21 @@
+import {visit} from 'unist-util-visit-esm';
+import {toString} from 'mdast-util-to-string';
+
+// eslint-disable-next-line unicorn/consistent-function-scoping
+const transform = () => (tree) => {
+  const headings = [];
+
+  visit(tree, `heading`, (heading) => {
```

Comment:
quibble: can we get some comments on the purpose of this plugin for future maintainers and how it's loaded.